website/docs/externaldata.md

```diff
@@ -0,0 +1,164 @@
+---
+id: externaldata
+title: External Data
+---
+
+> This feature is still in alpha stage, so the final form can still change (feedback is welcome!). Mutation is not yet supported with external data.
+
+## Motivation
+
+Gatekeeper provides various means to mutate and validate Kubernetes resources. However, in many of these scenarios this data is either built-in, static or user defined. With external data feature, we are enabling Gatekeeper to interface with various external data sources, such as image registries, using a provider-based model.
+
+A similar way to connect with an external data source can be done today using OPA's built-in `http.send` functionality. However, there are limitations to this approach.
+- Gatekeeper does not support Rego policies for mutation, which cannot use the OPA `http.send` built-in function.
+- Security concerns due to:
+  - if template authors are not trusted, it will potentially give template authors access to the in-cluster network.
+  - if template authors are trusted, authors will need to be careful on how rego is written to avoid injection attacks.
+
+Using the external data solution provides benefits, such as:
+- Addresses security concerns by:
+  - Restricting which hosts a user can access.
+  - Providing an interface for making requests, which allows us to better handle things like escaping strings.
+- Addresses common scenarios with a single provider, such as image SHA mutation that many of these scenarios may utilize.
```

Comment:
Are we using the word "scenarios" in two different ways on this bullet?

Reply:
updated